### ISSUE TITLE
_convert_timestamp assumes passed datetime to be UTC

### ISSUE THREAD

**Issue by gooney47** [influxdata/influxdb-client-python#73]
This function here https://github.com/influxdata/influxdb-client-python/blob/62f93503044dbbd86443555ccfac5cfd8359f91a/influxdb_client/client/write/point.py#L130 assumes a passed datetime object to be in UTC format if no timezone is specified. I find this to be highly misleading. From the moment on I set the local timezone when I install an OS I want my system to accept local timezone data from me if I don't specify otherwise.

**Comment by gooney47** [comment]
Note that also strings are assumed to be in UTC.

**Comment by bednar** [comment]
Hi @gooney47,

Thanks for using our client.

The InfluxDB 2 requires timestamp as UTC so we assume that a naive object `datetime(2009, 11, 10, 23, 0, 0, 123456)` is also in UTC. 

These data point produce same LineProtocol:

```python
Point.measurement("a")
     .field("val", 1)
     .time(datetime(2009, 11, 10, 23, 0, 0, 123456))
     .to_line_protocol()
```
```
a val=1i 1257894000123456000
```

```python
Point.measurement("a")
     .field("val", 1)
     .time(1257894000123456000)
     .to_line_protocol()
```
```
a val=1i 1257894000123456000
```

If you want to use a OS timezone use something like that: `datetime(2009, 11, 10, 23, 0, 0, 123456).astimezone(tzlocal())`.

Regards

**Comment by gooney47** [comment]
I still find this highly misleading for me for instance I was wondering why I couldn't see any data after I added some with giving datetime with local time (no timezone specified).  My timezone is +1. Now When I did a range query I couldn't see my data until I just for the sake of trial and error put a stop: 1d (which makes it so that the time range isn't stopped at the current time). The error could have been anywhere in my code at that point and that was just pure guessing since there were no errors.

Let's take a look at how datetimes should be worked with:
"With sufficient knowledge of applicable algorithmic and political time adjustments, such as time zone and daylight saving time information, an aware object can locate itself relative to other aware objects. An aware object represents a specific moment in time that is not open to interpretation. 1

A naive object does not contain enough information to unambiguously locate itself relative to other date/time objects. Whether a naive object represents Coordinated Universal Time (UTC), local time, or time in some other timezone is purely up to the program, just like it is up to the program whether a particular number represents metres, miles, or mass. Naive objects are easy to understand and to work with, at the cost of ignoring some aspects of reality." ([Source](https://docs.python.org/3/library/datetime.html))

This means that naive objects should not be assumed as UTC times, they can really be anything. The proper solution here - in my opinion - is raising an exception if no timezone is specified. Just adding the information into the documentation does not fix the problem - at least in my case - as I wasn't checking the documentation of that function specifically as the error could have been anywhere (database setup, not understanding the query language and what not...).

**Comment by bednar** [comment]
@gooney47 thanks for detail info

**Comment by bednar** [comment]
The default timezone can be changed by:

```python
from dateutil.tz import tzlocal

from influxdb_client import InfluxDBClient
from influxdb_client.client.util import date_utils
from influxdb_client.client.util.date_utils import DateHelper

date_utils.date_helper = DateHelper(timezone=tzlocal())

with InfluxDBClient(url="http://localhost:8086", token="my-token", org="my-org", debug=False) as client:
    ...

```

The issue is fixed in milestone [1.6.0](https://github.com/influxdata/influxdb-client-python/milestone/8).

If you would like to use dev version then install client by:

```console
pip install git+https://github.com/influxdata/influxdb-client-python.git@master
```
